src/components/sbb-timetable-row/sbb-timetable-row.helper.tsx

```diff
@@ -96,3 +106,94 @@ export const renderStringProduct = (vehicleName: string, line: string): JSX.Elem
     </span>
   );
 };
+
+const isReachable = (legs: PTRideLeg[]): boolean => {
+  return legs?.some((leg) => leg.serviceJourney?.serviceAlteration?.reachable === true);
+};
+
+const isRedirected = (legs: PTRideLeg[]): boolean => {
+  return legs?.some((leg) => leg.serviceJourney?.serviceAlteration?.redirected === true);
+};
+
+const isUnplannedStop = (legs: PTRideLeg[]): boolean => {
+  return legs?.some(
+    (leg) => !!leg.serviceJourney?.serviceAlteration?.unplannedStopPointsText === true
+  );
+};
+
+export const sortSituation = (situations: PtSituation[]): PtSituation[] => {
+  const priorities = {
+    DISTURBANCE: 0,
+    INFORMATION: 1,
+    DELAY: 2,
+    TRAIN_REPLACEMENT_BY_BUS: 3,
+    CONSTRUCTION_SITE: 4,
+    END_MESSAGE: 5,
+  };
+
+  return situations
+    ?.slice()
+    ?.sort((a: PtSituation, b: PtSituation) => priorities[a?.cause] - priorities[b?.cause]);
+};
+
+export const getHimIcon = (situation: PtSituation): string => {
+  switch (situation?.cause) {
+    case 'DISTURBANCE':
+      return 'disruption';
+    case 'INFORMATION':
+      return 'info';
+    case 'DELAY':
+      return 'delay';
+    case 'TRAIN_REPLACEMENT_BY_BUS':
+      return 'replacementbus';
+    case 'CONSTRUCTION_SITE':
+      return 'construction';
+    default:
+      return 'info';
+  }
+};
+
+export const getCus = (trip: Trip): string[] => {
+  const cus = [];
+  (trip?.summary.tripStatus?.cancelled || trip?.summary.tripStatus?.partiallyCancelled) &&
+    cus.push('cancellation');
+  !cus.length && !isReachable(trip?.legs) && cus.push('missed-connection');
+  !cus.length && trip?.summary?.tripStatus?.alternative && cus.push('alternative');
+  !cus.length && isRedirected(trip?.legs) && cus.push('reroute');
+  !cus.length && isUnplannedStop(trip?.legs) && cus.push('add-stop');
+  !cus.length &&
+    (trip?.summary?.tripStatus?.delayed || trip?.summary?.tripStatus?.delayedUnknown) &&
+    cus.push('delay');
+  !cus.length && trip?.summary?.tripStatus?.quayChanged && cus.push('platform-change');
+
+  return cus;
+};
+
+export const handleNotices = (notices: Notice[]): string[] => {
+  const sortedNotices = [];
+  const reservationNotice = [
+    'RR',
+    'RK',
+    'RC',
+    'RL',
+    'RM',
+    'RN',
+    'RS',
+    'RT',
+    'RU',
+    'XP',
+    'XR',
+    'XT',
+    'R',
+    'RB',
+  ];
+
+  notices?.some((notice) => notice.name === 'Z') && sortedNotices.push('sa-z');
```

Comment:
Not good to query the notices each time, lets hop on a call and see how we can improve this.